Makefile

```diff
@@ -0,0 +1,79 @@
+# temp storage of target
+TARGET=$(shell cat TARGET)
+
+# define the directories in which to search for library files
+PATH_LIBS =
+
+# define the directories in which to search for include files
+PATH_INCLUDE = -I include
+PATH_SRC = source
+PATH_EX = examples
+
+# define the library files
+FILES_LIBS = 
+
+# define the source & object files (most cases)
+FILES_SRC = $(PATH_EX)/$(TARGET)/ex-1.c\
+	$(PATH_SRC)/$(TARGET).c \
+	$(PATH_SRC)/utils.c
+         
+FILES_OBJ = $(PATH_EX)/$(TARGET)/ex-1.o\
+         $(PATH_SRC)/$(TARGET).o \
+         $(PATH_SRC)/utils.o
+
+# partial Makefiles will override for special cases
```

Comment:
This is really cool! Interested to learn more about how the partial makefiles work